src/main/scala/com/evolution/scache/LoadingCache.scala

```diff
@@ -116,110 +119,130 @@ private[scache] object LoadingCache {
+                                          case (state: EntryState.Loading[F, V], set) if state.deferred == deferred =>
+                                            set(EntryState.Value(entry))
                                               .flatMap {
-                                                case true  =>
+                                                // Happy path: successfully placed our computed value
+                                                case true =>
                                                   a
                                                     .asLeft[Either[F[V], V]]
                                                     .pure[F]
+                                                // Failed to set our value, so we just release it and return:
+                                                // - Another value (computed or ongoing), if found in the ref
+                                                // - Our value otherwise
                                                 case false =>
-                                                  deferred
-                                                    .getOrError
-                                                    .map { entry =>
-                                                      entry
-                                                        .value
-                                                        .asRight[F[V]]
-                                                        .asRight[A]
+                                                  entry
+                                                    .release1
+                                                    .start
+                                                    .flatMap { _ =>
+                                                      entryRef
+                                                        .optEither
+                                                        .map(_.toRight(a))
                                                     }
                                               }
-                                              .map { _.asRight[Int] }
-                                          case false =>
-                                            (counter + 1)
-                                              .asLeft[Either[A, Either[F[V], V]]]
-                                              .pure[F]
+
+                                          // Entry state got updated while we were loading, so we release our value
+                                          // and return a value based on the latest entry content.
+                                          case (state, _) =>
+                                            entry
+                                              .release1
+                                              .start
+                                              .as {
+                                                state
+                                                  .optEither
```

Comment:
should we `retry` here upon `Removed` ?

Reply:
If we encounter `Removed`, the contents of our `entryRef` won't ever change anymore;
The only reason to `retry` would be if we looked up the key in the map again: `ref.get.map(_.get(key))...`

Reply:
the problem I want to fix - that I do not want to return back an error from `getOrUpdate` function upon conflict when the cell has `Removed` inside.